### PR TITLE
fix(ticket): approvers viewing shadow table sync ticket fails after the ticket is approved/rejected

### LINKED ISSUE #1048: [Bug]: After successfully applying to execute the shadow table, an error message appears when viewing tasks for the admin user
### ODC version

odc423

### OB version

*

### What happened?

image

### What did you expect to happen?

Can be viewed

### How can we reproduce it (as minimally and precisely as possible)?

1. Performing shadow table synchronization on low permission accounts
2. Agree with Damin account
3. View shadow table synchronization execution tasks

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

type-bug

#### What this PR does / why we need it:
Shadow table sync tickets should be seen by creators and approvers but after the tickets are approved by the approval, it would occur an error when the approver clicks the ticket again.
The root cause is that when we query approval candidates, we use 'EXECUTING' status to filter the approval instances, but not filter the 'APPROVED' approval instances.
This PR changed the method signature and could list both approving and approved approval instances.

#### Which issue(s) this PR fixes:

Fixes #1048 

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/ApprovalPermissionService.java [hunk 1]
```diff
@@ -83,60 +76,6 @@ public class ApprovalPermissionService {
...
     private UserResourceRoleRepository userResourceRoleRepository;
 
-    /**
-     * 获取审批节点可审批用户集合
-     *
-     * @param approvalInstanceId
-     * @param enabled 关联用户或角色是否启用，true -> 角色用户启用; false -> 角色用户禁用; null -> 启用禁用用户均筛选出来
-     * @return list of {@link User}
-     */
-    public Set<User> getCandidates(@NonNull Long approvalInstanceId, Boolean enabled) {
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/ApprovalPermissionService.java:93]: Delete the deprecated code without any reference. Same as follows.

**Comment by yhilmare** [comment]
so the related issue is not only for shadow table task but also all types of task?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/ApprovalPermissionService.java [hunk 2]
```diff
@@ -256,18 +130,27 @@ public List<UserTaskInstanceEntity> listApprovableExternalInstances() {
                 .collect(Collectors.toList());
     }
 
+
+    public Map<Long, Set<UserEntity>> getApproverByFlowInstanceIds(@NonNull Collection<Long> flowInstanceIds) {
+        return getUsersByFlowInstanceIdsAndStatus(flowInstanceIds, FlowNodeStatus.getExecutingAndFinalStatuses());
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/ApprovalPermissionService.java:135]: prefer `return getUsersByFlowInstanceIdsAndStatus(flowInstanceIds, Collectors.sington(FlowNodeStatus.EXECUTING));`

**Comment by MarkPotato777** [comment]
> so the related issue is not only for shadow table task but also all types of task?

No, it's just for shadow table sync tasks which is a kind of special. Shadow table sync tasks are not flowable tasks, and they have a separate table to record. Here it uses related flow_instance_id to query all approvers to make them have authority to view the shadow table sync results.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/ApprovalPermissionService.java (hunk 2 above)
Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/ApprovalPermissionService.java:135]: I need exactly 'executing' and 'final' status here for querying those approvers who are approving, **approved/rejected**, and this is where the bug is located.

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**